sample_files/.travis.yml

```diff
@@ -27,6 +26,42 @@ addons:
 #  - gem install bootstrap-sass
 #  - gem install compass --pre
 
+stages:
+  - linting
+  - test
+
+jobs:
```

Comment:
FYI I need improve travis2docker in order to support this new way of `.travis.yml`
Meanwhile runbot won't work (notice skipped status).

Reply:
Ah, I hope it's not a too big change.

I had to make it this way to run pre-commit with python 3.6 while still running the Odoo tests with 3.5.

As a side benefit, we avoid running test jobs when linting fails, which saves us so Travis workers.

Reply:
Ok, the fancy travis report output is so good too.
hahaha
Let me check it...

Reply:
AFAIK, now Odoo 13 main Python version is 3.6 (maybe we need to change this massively).

Reply:
> Odoo 13 main Python version is 3.6

Are you sure? I got several echoes during OdooXP it was still 3.5.

Reply:
> maybe we need to change this massively

In any case I was planning to reset all travis configs to default on 13.0.
It's a good occasion to realign everything and people can easily forward port per-repo specific adaptations when needed.

Reply:
We need to be 100% sure about that. As much as I'd love to use 3.6 features, we don't want to have OCA modules don't work on some Odoo deployments.

@mart-e @odony can one of you confirm what is the minimum supported python version for Odoo 13?

Reply:
Odoo 13' setup.py also `says >= 3.5`

Reply:
Debian 10 ships with 3.7.

This debate is not very important. I think it comes from an outdated mindset that more or less we all have inside. We have been forced to stick with python 2.7 for decades, so the versioning thing was like carved on stone in our brains.

As we make the leap for python 3.5+, it's possible, maintainable and easy to write code that works fine across 3.5, 3.6, 3.7 and 3.8. Just avoid writing `async` and `await` and you're done, basically.

So my vote goes for using the version that means less problems and maintenance. If it is 3.6, then go for it. :blush: 

<details><summary>:warning: off-topic content ahead :warning: </summary>

IMHO there's a conceptual mistake in the travis2docker thing (I already spoke about this with some of you in 2018). It's really a lot easier to do it reverse: just make travis use docker, instead of make docker imitate travis. This way, each base image for each odoo version includes the dependencies it needs, and all just works. Travis uses py3.6+ to lint, and the underlying image uses whatever python version is shipped within it. That'd be much easier.

Without trying to impose our solution, just FTR doodba has 2 different set of tags. I.e., image `tecnativa/doodba:12.0-onbuild` includes all the "doodba experience magic", but also there is `tecnativa/doodba:12.0` which only includes all dependencies needed by odoo 12.0 in a stable base. Those can be used for MQT and would make life easier for Travis and, most important, its maintainers. I'm pretty sure that all of you have similar solutions. We should flip that switch and delete travis2docker.

Back to the topic. :sweat_smile: 
</details>

Reply:
@Yajo regarding the python version, python 3 backward compatibility can be taken for granted for sure, that is not the issue. We want OCA modules to work wherever Odoo is deployed. If Odoo says the minimum supported version is 3.5, some people somewhere will deploy it on that version and we need to test on 3.5 *to make sure OCA addons don't use 3.6+ features* that would break in some places where Odoo otherwise works fine.

Reply:
BTW, @moylop260 whatever the answer is regarding the python version, the travis2docker update is useful, because there are Travis workload benefits in doing linting in a separate job.

Reply:
> Are you sure? I got several echoes during OdooXP it was still 3.5.

@sbidoul Would you by any chance recall who mentioned 3.5?

We haven't changed the minimum version yet, and we globally refrained from using new features like f-strings. Yet I'm not sure we will be able to keep supporting 3.5 officially. Our 13.0 packages are all built using 3.6 or 3.7, and the runbot Dockerfile is using a single image based on Bionic so 3.6.

We've just tried, the 13.0 testsuite does not pass with 3.5 anymore. It's mostly minor things (dict key order, etc.), but it will keep breaking if we don't test with 3.5. This, plus the fact that our benchmark distros both have 3.6+ now, makes a good case for 3.6.
We'll double-check internally and @d-fence will probably update the minimum version later today or tomorrow.

Reply:
> Would you by any chance recall who mentioned 3.5?

Not really. I think I chatted about it with @mart-e and got an indirect mention of it by a colleague. Was it @lmignon ?

@moylop260 @gurneyalex will we need to do something for the OCA runbot to use python 3.6 on the 13.0 branch?

Reply:
@sbidoul 
We don't need to do anything extra to use python3.6 for OCA runbot.
It is pre-installed py3.5, py3.6 and py3.7 and it is used from `.travis.yml`

Reply:
https://github.com/odoo/odoo/pull/38364 is approved, to 3.6 it is.